Tribler/Core/Modules/restapi/my_channel_endpoint.py

```diff
@@ -120,6 +123,71 @@ def render_PUT(self, request):
         return json.dumps({"added": channel_id})
 
 
+class MyChannelCreateTorrentEndpoint(MyChannelBaseEndpoint):
+    """
+    Create a torrent file from local files.
+    See: http://www.bittornado.com/docs/multitracker-spec.txt
+
+    Example request:
+    {
+        "files": ['path/to/file.txt', 'path/to/another/file.mp4'],
+        "tracker": "http://localhost/announce",
+    (or)"tracker-list": [ [tracker1], [backup1], [backup2] ]
+        "name": "Cat video",
+        "description: "Video of my cat"
+    }
+    """
+
+    def render_GET(self, request):
+        """
+        This method creates a torrent from local files and returns it.
+        Note that the source files need to exist and are skipped otherwise.
+        """
+        parameters = http.parse_qs(request.content.read(), 1)
+        tdef = TorrentDef()
+
+        try:
+            if 'files' in parameters and len(parameters['files']) > 0:
+                files = ast.literal_eval(parameters['files'][0])
+                for source_file in files:
+                    if os.path.exists(source_file["uri"]):
+                        tdef.add_content(source_file["uri"])
+                    else:
+                        raise ValueError("File does not exist: " + source_file["uri"])
+            else:
+                request.setResponseCode(http.BAD_REQUEST)
+                return json.dumps({"error": "files parameter missing"})
+
+            if 'tracker' in parameters and len(parameters['tracker']) > 0:
+                tdef.set_tracker(parameters['tracker'][0])
+            elif 'tracker-list' in parameters and len(parameters['tracker-list']) > 0:
+                tdef.set_tracker_hierarchy(parameters['tracker-list'][0])
+            else:
+                request.setResponseCode(http.BAD_REQUEST)
+                return json.dumps({"error": "tracker(-list) parameter missing"})
+
+            if 'name' in parameters and len(parameters['name']) > 0:
+                tdef.set_name(parameters['name'][0])
+
+            if 'description' in parameters and len(parameters['description']) > 0:
+                tdef.set_comment(['description'][0])
+
+        except ValueError as ex:
```

Comment:
I intended to catch all ValueErrors thrown by the methods of TorrentDef that are called here, because any value given by the user as input to TorrentDef is not a crash of Tribler and therefore should be caught and handled to send back to the GUI, as designed for handled errors.